### PR TITLE
Refactor BooleanBuilder Constructors

### PR DESCRIPTION
**Pull request by psvri**
# Which issue does this PR close?

Partially implements https://github.com/apache/arrow-rs/issues/2054

# Rationale for this change
 
Refactor apis to make a consistent new and with_capacity functions

# What changes are included in this PR?

Removes capacity from new and added with_capacity function instead.

# Are there any user-facing changes?

Yes

### CODE REVIEW

File: arrow/src/array/builder/boolean_builder.rs [hunk 1]
```diff
@@ -67,8 +67,14 @@ pub struct BooleanBuilder {
 }
 
 impl BooleanBuilder {
-    /// Creates a new primitive array builder
-    pub fn new(capacity: usize) -> Self {
+    /// Creates a new boolean builder
+    #[allow(clippy::new_without_default)]
```
Comment (tustvold) [inline arrow/src/array/builder/boolean_builder.rs:71]: I think we probably should define Default

File: arrow/src/array/builder/boolean_builder.rs [hunk 2]
```diff
@@ -67,8 +67,14 @@ pub struct BooleanBuilder {
...
+    pub fn new() -> Self {
+        Self::with_capacity(1024)
+    }
+
+    /// Creates a new boolean builder with capacity
```
Comment (tustvold) [inline arrow/src/array/builder/boolean_builder.rs:76]:
```suggestion
    /// Creates a new boolean builder with space for `capacity` elements without re-allocating
```

**Review by tustvold — APPROVED** [review]

File: arrow/src/array/builder/boolean_builder.rs (hunk 1 above)
Comment (psvri) [reply arrow/src/array/builder/boolean_builder.rs:71]: I too was thinking the same , but then other builders didnt have default, so to keep them consistent I had taken that decision. 

Let me know your thoughts.

Comment (tustvold) [reply arrow/src/array/builder/boolean_builder.rs:71]: I think there isn't a harm in defining it where it is possible, I think the fixed length builders are the only ones where it isn't possible

Comment (psvri) [reply arrow/src/array/builder/boolean_builder.rs:71]: Okay, I shall add them

File: arrow/src/array/builder/boolean_builder.rs (hunk 2 above)
Comment (psvri) [reply arrow/src/array/builder/boolean_builder.rs]: Done in latest commit

File: arrow/src/array/builder/boolean_builder.rs (hunk 1 above)
Comment (psvri) [reply arrow/src/array/builder/boolean_builder.rs]: Done in latest commit

**Pull request merged**